src/common/CMakeLists.txt

```diff
@@ -4,5 +4,8 @@ facelift_add_library(FaceliftCommonLib
     HEADERS
         span.h
         FaceliftUtils.h
+    HEADERS_NO_MOC
```

Comment:
How does HEADERS_NO_INSTALL work with HEADERS_NO_MOC ?

Reply:
Nohow, I'll take care of it.

Reply:
But no, it is works like this:
HEADERS_NO_MOC are excluded from both HEADERS and HEADERS_NO_INSTALL.
Is this the correct way ?

Reply:
IMO this is correct way. If we do not want to run moc and don't want to install it, we do not need to add the header at all.

Reply:
we need to provide a way to install headers which do not need the moc.

HEADERS_NO_INSTALL adds new headers (whitelist) whereas your implementation of HEADERS_NO_MOC removes (blacklist) content from list of headers. This is inconsistent.

Reply:
Actually HEADERS_NO_MOC should install headers. @kunichik please test if it works this way.
Summary:
- HEADERS -> run moc, install
- HEADERS_NO_INSTALL -> run moc, don't install
- HEADERS_NO_MOC -> don't run moc, install
- header not added -> don't run moc, don't install (no need to add header like that)

Reply:
yep, I've not added that
HEADERS_NO_MOC are excluded from both HEADERS and HEADERS_NO_INSTALL and write to the new variable HEADERS_FOR_MOC. The variable sends as argument for moc.
HEADERS and HEADERS_NO_INSTALL themself don't change at all.

> HEADERS -> run moc, install
true
> HEADERS_NO_INSTALL -> run moc,
true, but
>  don't install
Except of adding to moc processing, HEADERS_NO_INSTALL don't use anywhere.
In the previous realization HEADERS_NO_INSTALL had been only added to the moc processing:
`set(HEADERS_FOR_MOC ${HEADERS} ${HEADERS_NO_INSTALL})`
 that's it
    
> HEADERS_NO_MOC -> don't run moc, install
true for those headers, that are in HEADERS.
HEADERS_NO_MOC is only removes headers from HEADERS and HEADERS_NO_INSTALL

>  header not added -> don't run moc, don't install (no need to add header like that)
true